### PR TITLE
sleep: don't init /sys/power/resume if 'resume=' option is missing an…

### PR DESCRIPTION
**Pull request by fbuihuu**
…d EFI is disabled

Otherwise in such case a first `systemctl hibernate` would fail but would still initialize /sys/power/resume fooling a second `systemctl hibernate` into believing that 'resume=' is correctly set and can be used by the resume process to find the swap device to resume from.

Follow-up for #27330.

### CODE REVIEW

File: src/sleep/sleep.c [hunk 1]
```diff
@@ -262,16 +262,18 @@ static int execute(
                         return log_error_errno(r, "Failed to find location to hibernate to: %m");
                 resume_set = r > 0;
 
+                r = write_efi_hibernate_location(hibernate_location, !resume_set);
                 if (!resume_set) {
+                        if (r == -EOPNOTSUPP)
+                                return log_error_errno(r, "Failed to hibernate as resuming without a valid 'resume=' option won't work");
```
Comment (YHNdnzj) [inline src/sleep/sleep.c]: Maybe rephrase this as "Refusing hibernation, resuming isn't possible as 'resume=' is not set."?

(Since it's not actually a "failure" for hibernation.)

**Review by YHNdnzj — APPROVED** [review]

**Comment by keszybz** [comment]
LGTM too.

File: src/sleep/sleep.c (hunk 1 above)
Comment (fbuihuu) [reply src/sleep/sleep.c]: I updated the message with: "No valid 'resume=' option found, refusing to hibernate"

Comment (YHNdnzj) [reply src/sleep/sleep.c]: Please add a period at the end of the sentence.

**Review by bluca — APPROVED** [review]
